qiskit_experiments/analysis/curve_fitting.py

```diff
@@ -213,7 +214,7 @@ def multi_curve_fit(
     weights: Optional[np.ndarray] = None,
     bounds: Optional[Union[Dict[str, Tuple[float, float]], Tuple[np.ndarray, np.ndarray]]] = None,
     **kwargs,
-) -> CurveAnalysisResult:
+) -> Dict:
```

Comment:
I prefer to have meaningful type annotation rather than `Dict`, i.e. I want to make sure this output is used for curve analysis and some visualization functions without reading the codebase. This makes data structure clearer, i.e. visualization function can know the data has key such as `popt`, `popt_kyes`, ...

Reply:
So I created a `CurveAnalysisResultData` in 0d6e3c9. I tried to add some keys to it (`popt`, `popt_kyes`, `xrange`, etc) with initial values of `None`. But that caused `test_run_single_curve_fail` to fail because in `CurveAnalysis._run_analysis`, it still tries to create a plot even when analysis failed, and it choked on the `None` value. I decided I don't understand that path well enough so I just left `CurveAnalysisResultData` as an empty dictionary 😅

Reply:
Thanks @jyu00 I think this looks good for now :)